signer/threshold_validator.go

```diff
@@ -457,44 +567,13 @@ func (pv *ThresholdValidator) SignBlock(chainID string, block *Block) ([]byte, t
 	}
 
 	// Keep track of the last block that we began the signing process for. Only allow one attempt per block
-	if err := pv.SaveLastSignedStateInitiated(chainID, NewSignStateConsensus(height, round, step)); err != nil {
-		switch err.(type) {
-		case *SameHRSError:
-			// Wait for last sign state signature to be the same block
-			signAgain := false
-			for i := 0; i < 100; i++ {
-				existingSignature, existingTimestamp, sameBlockErr := pv.getExistingBlockSignature(chainID, block)
-				if sameBlockErr == nil {
-					if existingSignature == nil {
-						signAgain = true
-						break
-					}
-					return existingSignature, existingTimestamp, nil
-				}
-				switch sameBlockErr.(type) {
-				case *StillWaitingForBlockError:
-					time.Sleep(10 * time.Millisecond)
-					continue
-				default:
-					return nil, existingTimestamp, sameBlockErr
-				}
-
-			}
-			if !signAgain {
-				return nil, stamp, errors.New("timed out waiting for block signature from cluster")
-			}
-		default:
-			existingSignature, existingTimestamp, sameBlockErr := pv.getExistingBlockSignature(chainID, block)
-			if sameBlockErr == nil {
-				return existingSignature, stamp, nil
-			}
-			hrs := HRSKey{
-				Height: height,
-				Round:  round,
-				Step:   step,
-			}
-			return nil, existingTimestamp, pv.newBeyondBlockError(chainID, hrs)
-		}
+	existingSignature, existingTimestamp, err := pv.SaveLastSignedStateInitiated(chainID, block)
+	if err != nil {
+		return nil, stamp, err
+	}
+	if existingSignature != nil {
+		pv.logger.Debug("Returning existing signature", "signature", fmt.Sprintf("%x", existingSignature))
```

Comment:
Note here, the `fmt.Sprintf` call will be eagerly evaluated, regardless of whether the logger is debug-level enabled.

If this is an infrequently executed code path -- and I think that's the case here -- that is fine.